### PR TITLE
Add haxe.Constraints.NotVoid

### LINKED ISSUE #6810: "Non-Void" constraint for type parameters
I'd like to have a "Non-Void" constraint for type parameters. 
Here is why:
```haxe
class Test {
    static function main() {
      	var a = [1, 2, 3].map(function(i) i * 2)); //Compiler does not complain that function does not returning any value.
        trace(a); //null, null, null
    }
}
```
https://try.haxe.org/#a296c

### PR DESCRIPTION
**Pull request by kLabz**
Please tell me if this should be handled somewhere else too and/or if there are ways to test it that I missed.

See #6810

### CODE REVIEW

File: src/core/type.ml [hunk 1]
```diff
@@ -2017,6 +2017,9 @@ let rec unify a b =
 	| TAbstract ({a_path=[],"Void"},_) , _
 	| _ , TAbstract ({a_path=[],"Void"},_) ->
 		error [cannot_unify a b]
+    | TAbstract ({ a_path = ["haxe"],"NotVoid" },[]), _
```
Comment (RealyUniqueName) [inline src/core/type.ml]: Please, change spaces to tabs as in the neighbring lines.

Comment (kLabz) [reply src/core/type.ml]: O_o sorry, seems like my editor is not configured correctly on this computer..

**Comment by RealyUniqueName** [comment]
As #6810 is still in "Design" milestone, we need to make a decision there before merging this.

**Comment by Simn** [comment]
#6810 is still under discussion, but this PR needs an update regardless.

**Comment by RealyUniqueName** [comment]
> this PR needs an update regardless

This PR has no conflicts currently.

**Comment by kLabz** [comment]
Yeah I updated my PRs

**Comment by Simn** [comment]
#6810 is a bit tricky to implement so I think this PR is a good (temporary?) solution that won't cause us any trouble in the long run. Any `NotVoid` constraint being used now is still going to be fine in the future, so I'm in favor of this addition.

**Pull request merged**